agent/check_handler.go

```diff
@@ -89,10 +90,13 @@ func (a *Agent) executeCheck(request *v2.CheckRequest) {
 		return
 	}
 
+	// Prepare environment variables
+	env := environment.MergeEnvironments(syscall.Environ(), assets.Env(), check.EnvVars)
```

Comment:
I would use `os.Environ()` rather than `syscall.Environ()`, even if right now `os.Environ()` simply calls `syscall.Environ()`.

The `syscall` package is meant to be a low level interface; using the `os` package is preferred in most cases.